MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectJsonMenuComponent.cpp

```diff
@@ -0,0 +1,190 @@
+    {
+        m_onConnectToHostFailed.Disconnect();
+        UiCursorBus::Broadcast(&UiCursorInterface::DecrementVisibleCounter);
+
+        Multiplayer::SessionAsyncRequestNotificationBus::Handler::BusDisconnect();
+    }
+
+    void UiGameLiftConnectJsonMenuComponent::OnJSONChanged(const AZStd::string& gameLiftJsonString)
+    {
+        // Disable the connect button until checking to make sure the user has provided the proper GameLift information in JSON format
+        UiInteractableBus::Event(m_connectButtonUi, &UiInteractableInterface::SetIsHandlingEvents, false);
+
+        if (gameLiftJsonString.empty())
+        {
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Please provide GameLift JSON!");
+            return;
+        }
+        
+        // Parse GameLift JSON
+        m_request = {};
+        rapidjson::Document document;
+        document.Parse(gameLiftJsonString.c_str());
+
+        if (document.HasParseError())
+        {
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Invalid JSON format!");
+            return;
+        }
+
+        if (!document.HasMember("GameSessionId"))
+        {
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Missing GameSessionId!");
+            return;
+        }
+
+        if (!document.HasMember("PlayerSessionId"))
+        {
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Missing PlayerSessionId!");
+            return;
+        }
+
+        const rapidjson::Value& gameSessionId = document["GameSessionId"];
+        if (!gameSessionId.IsString())
+        {
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Invalid GameSessionId!");
+            return;
+        }
+
+        const rapidjson::Value& playerSessionId = document["PlayerSessionId"];
+        if (!playerSessionId.IsString())
+        {
+            UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "Invalid PlayerSessionId!");
+            return;
+        }
+
+        UiTextBus::Event(m_jsonParseFailTextUi, &UiTextInterface::SetText, "");
+        m_request.m_sessionId = gameSessionId.GetString();
+        m_request.m_playerId = playerSessionId.GetString();
+        UiInteractableBus::Event(m_connectButtonUi, &UiInteractableInterface::SetIsHandlingEvents, true);
+    }
+
+    void UiGameLiftConnectJsonMenuComponent::OnButtonClicked(AZ::EntityId buttonEntityId) const
+    {
+        const auto console = AZ::Interface<AZ::IConsole>::Get();
+        if (!console)
+        {
+            AZ_Assert(false, "UiGameLiftConnectJsonMenuComponent attempting to use console commands before AZ::Console is available.");
+            return;
+        }
+
+        if (buttonEntityId == m_quitButtonUi)
+        {
+            console->PerformCommand("quit");
+        }
+        else if (buttonEntityId == m_connectButtonUi)
```

Comment:
could we `return;` at the end of the previous block, and make this a regular `if` statement instead of an if/else? (same goes for the following if/else block)

Reply:
Fixed